### ISSUE TITLE
Wrong git branch name in edge case.

### ISSUE THREAD

**Issue by Carreau** [xonsh/xonsh#1778]
```
path/to/repo master $ git checkout origin/pr/132
path/to/repo origin/pr/132 $ # do changes
path/to/repo origin/pr/132 $ git commit -am 'stuff'
...
path/to/repo from $ #WTF ? Why "from" branch ?
path/to/repo from $ git branch
* (HEAD detached from origin/pr/132)
  106
  community-page
  fix-css
  foo
  master
```

Oh, make sens.

**Comment by laerus** [comment]
Sorry about this @Carreau, on it!

**Comment by Carreau** [comment]
> Sorry about this @Carreau, on it!

No need to apologize. I can likely have a shot at it but I wanted to open the issue before forgetting.

**Comment by laerus** [comment]
i introduced this bug recently, potential fix in #1779
